database-migration/src/main/java/uk/ac/ebi/eva/dbmigration/mongodb/ExtractStatisticsFromVariant.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbmigration.mongodb;
+
+import com.github.mongobee.changeset.ChangeLog;
+import com.github.mongobee.changeset.ChangeSet;
+import com.mongodb.BasicDBObject;
+import com.mongodb.BulkWriteOperation;
+import com.mongodb.DBCollection;
+import com.mongodb.DBCursor;
+import com.mongodb.DBObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.mongodb.core.MongoTemplate;
+
+/**
+ * Script that executes the following steps using mongobee (https://github.com/mongobee/mongobee/wiki/How-to-use-mongobee):
+ * - Extracts the 'st' field from a Variant stored in MongoDB into a new statistics object
+ * - _id, chr, start, end, ref and alt are added into the Statistics object
+ * - The Statistics object is added into the statistics collection
+ * - The 'maf' field is added into the Variant object
+ * - The 'st' field is removed from the Variant object
+ */
+@ChangeLog
+public class ExtractStatisticsFromVariant {
+    private static final Logger logger = LoggerFactory.getLogger(ExtractStatisticsFromVariant.class);
+
+    private final static int BULK_SIZE = 1000;
+
+    @ChangeSet(order = "001", id = "migrateStatistics", author = "EVA")
+    public void migrateStatistics(MongoTemplate mongoTemplate) {
+        logger.info("1) migrate statistics {}", MongoMigrationMain.variantsCollectionName);
+
+        final DBCollection statisticsCollection = mongoTemplate
+                .getCollection(MongoMigrationMain.statisticsCollectionName);
+        final DBCollection variantsCollection = mongoTemplate.getCollection(MongoMigrationMain.variantsCollectionName);
+
+        BulkWriteOperation bulkInsertMaf = variantsCollection.initializeUnorderedBulkOperation();
+        BulkWriteOperation bulkInsertStats = statisticsCollection.initializeUnorderedBulkOperation();
+
+        int counter = 0;
+        DBCursor variantCursor = variantsCollection.find();
+
+        while (variantCursor.hasNext()) {
+            DBObject variantObj = variantCursor.next();
+
+            DBObject statsObj = (DBObject) variantObj.get("st");
+
+            if (statsObj != null) {
+                for (String standaloneStatsObj : statsObj.keySet()) {
+                    BasicDBObject iStatsObj = (BasicDBObject) statsObj.get(standaloneStatsObj);
```

Comment:
Please ensure the resulting documents match the structure of [the PopulationStatistics class](https://github.com/EBIvariation/eva-pipeline/blob/develop/src/main/java/uk/ac/ebi/eva/pipeline/model/PopulationStatistics.java). At least, the original statistics had a `fid` that will be no longer used.